### PR TITLE
Add test coverage for file repo functionality (BZ1446975)

### PR DESCRIPTION
**Pull request by oshtaier**
`+ few fixes`
```
nosetests tests/foreman/cli/test_repository.py -m file_repo
S...
----------------------------------------------------------------------
Ran 4 tests in 87.327s

OK (SKIP=1)
```

### CODE REVIEW

File: tests/foreman/cli/test_repository.py [hunk 1]
```diff
@@ -229,7 +262,7 @@ def test_positive_create_with_auth_yum_repo(self):
         url = FAKE_5_YUM_REPO
         for creds in valid_http_credentials(url_encoded=True):
             url_encoded = url.format(creds['login'], creds['pass'])
-            with self.subTest(url):
+            with self.subTest(creds):
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: cred is dict perhaps url_encoded should be more appropriate?

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -443,7 +476,7 @@ def test_positive_create_with_auth_puppet_repo(self):
         url = FAKE_7_PUPPET_REPO
         for creds in valid_http_credentials(url_encoded=True):
             url_encoded = url.format(creds['login'], creds['pass'])
-            with self.subTest(url):
+            with self.subTest(creds):
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: same question as above

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -171,7 +176,7 @@ def test_positive_create_with_name_label(self):
                     u'name': name,
                 })
                 self.assertEqual(new_repo['name'], name)
-                self.assertNotEqual(new_repo['name'], new_repo['label'])
+                self.assertEqual(new_repo['label'], label)
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: Please provide a test for this change

File: tests/foreman/cli/test_repository.py (hunk 1 above)
Comment (oshtaier) [reply tests/foreman/cli/test_repository.py]: but here
https://github.com/SatelliteQE/robottelo/pull/5691/files#diff-c5b3908e7803d076d19a827bf7c96c31R697
it is fine.
what is right way?

File: tests/foreman/cli/test_repository.py [hunk 4]
```diff
@@ -784,7 +821,30 @@ def test_positive_synchronize_yum_repo(self):
                 self.assertEqual(new_repo['sync']['status'], 'Success')
 
     @run_only_on('sat')
-    @skip_if_bug_open('bugzilla', 1328092)
+    @tier2
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: Think @tier1 is more appropriate, what do you think?

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_repository.py
ESSSSSSSSSSSSSSSSS..........S......E..........................................S.E
======================================================================
ERROR: test suite for <class 'tests.foreman.cli.test_repository.DRPMRepositoryTestCase'>
----------------------------------------------------------------------
======================================================================
ERROR: Create a Docker repository with a random name.
----------------------------------------------------------------------
======================================================================
ERROR: test suite for <class 'tests.foreman.cli.test_repository.SRPMRepositoryTestCase'>
----------------------------------------------------------------------
```

File: tests/foreman/cli/test_repository.py (hunk 4 above)
Comment (oshtaier) [reply tests/foreman/cli/test_repository.py]: It is only one component affected, so it sounds as tier1
ok, then I need to change for all other sync tests?

File: tests/foreman/cli/test_repository.py [hunk 5]
```diff
@@ -784,7 +821,30 @@ def test_positive_synchronize_yum_repo(self):
...
+            u'url': CUSTOM_FILE_REPO,
+        })
+        # Assertion that repo is not yet synced
+        self.assertEqual(new_repo['sync']['status'], 'Not Synced')
+        # Synchronize it
+        Repository.synchronize({'id': new_repo['id']})
+        # Verify it has finished
+        new_repo = Repository.info({'id': new_repo['id']})
+        self.assertEqual(new_repo['sync']['status'], 'Success')
+        self.assertEqual(new_repo['content-counts']['files'], '3')
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: please add a constant with the same name as this repo eg:
```python
CUSTOM_FILE_REPO_FILES_COUNT = 3
```

File: tests/foreman/cli/test_repository.py [hunk 6]
```diff
@@ -794,15 +854,19 @@ def test_positive_synchronize_auth_yum_repo(self):
...
-                      if cred['http_valid']]:
+        for creds in [
+            cred for cred in valid_http_credentials(url_encoded=True)
+            if cred['http_valid']
+        ]:
             url_encoded = url.format(
                 creds['login'], creds['pass']
             )
-            with self.subTest(url):
+            with self.subTest(creds):
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: same question as above

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_repository.py -m file_repo
S...
----------------------------------------------------------------------
Ran 4 tests in 90.451s

OK (SKIP=1)
```

File: tests/foreman/cli/test_repository.py [hunk 7]
```diff
@@ -662,8 +695,10 @@ def test_negative_create_with_auth_url_with_special_characters(self):
         :CaseImportance: Critical
         """
         # get a list of valid credentials without quoting them
-        for cred in [creds for creds in valid_http_credentials()
-                     if creds['quote'] is True]:
+        for cred in [
+            creds for creds in valid_http_credentials()
+            if creds['quote'] is True
+        ]:
```
Comment (ldjebran) [inline tests/foreman/cli/test_repository.py]: this also can be updated to 
```python
for cred in [
    creds for creds in valid_http_credentials()
    if creds['quote'] is True
]:
    url_encoded = FAKE_5_YUM_REPO.format(cred['login'], cred['pass'])		                  
    with self.subTest(url_encoded):
        with self.assertRaises(CLIFactoryError):
        ...

```

Comment (oshtaier) [reply tests/foreman/cli/test_repository.py]: sure, but all these changes were not my initial intention anyway=)
so it can be fixed in next PRs

Comment (ldjebran) [reply tests/foreman/cli/test_repository.py]: yes I know, but one more test line and we finish with all this refactoring for ever (I hope) :)

Comment (oshtaier) [reply tests/foreman/cli/test_repository.py]: that is applicable for two tests, not one, but no problem
Fixed

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_repository.py -m auth
.........
----------------------------------------------------------------------
Ran 9 tests in 1194.008s

OK
```

File: tests/foreman/cli/test_repository.py [hunk 8]
```diff
@@ -784,7 +820,33 @@ def test_positive_synchronize_yum_repo(self):
...
+        # Assertion that repo is not yet synced
+        self.assertEqual(new_repo['sync']['status'], 'Not Synced')
+        # Synchronize it
+        Repository.synchronize({'id': new_repo['id']})
+        # Verify it has finished
+        new_repo = Repository.info({'id': new_repo['id']})
+        self.assertEqual(new_repo['sync']['status'], 'Success')
+        self.assertEqual(
+            new_repo['content-counts']['files'],
+            str(CUSTOM_FILE_REPO_FILES_COUNT)
```
Comment (abalakh) [inline tests/foreman/cli/test_repository.py]: And if tomorrow devs fix output and files count becomes int value? Better solve it the opposite way:
```python
self.assertEqual(
    int(new_repo['content-counts']['files']),
    CUSTOM_FILE_REPO_FILES_COUNT
)
```

File: tests/foreman/cli/test_repository.py [hunk 9]
```diff
@@ -1402,6 +1468,49 @@ def test_positive_upload_content(self):
...
+        new_repo = self._make_repository({
+            u'content-type': u'file',
+            u'url': CUSTOM_FILE_REPO,
+        })
+        Repository.synchronize({'id': new_repo['id']})
+        # Verify it has finished
+        new_repo = Repository.info({'id': new_repo['id']})
+        self.assertEqual(
+            new_repo['content-counts']['files'],
+            str(CUSTOM_FILE_REPO_FILES_COUNT)
```
Comment (abalakh) [inline tests/foreman/cli/test_repository.py]: same here

File: tests/foreman/cli/test_repository.py [hunk 10]
```diff
@@ -1402,6 +1468,49 @@ def test_positive_upload_content(self):
...
+            'product-id': new_repo['product']['id'],
+        })
+        self.assertIn(
+            "Successfully uploaded file '{0}'".format(OS_TEMPLATE_DATA_FILE),
+            result[0]['message'],
+        )
+        new_repo = Repository.info({'id': new_repo['id']})
+        self.assertEqual(
+            new_repo['content-counts']['files'],
+            str(CUSTOM_FILE_REPO_FILES_COUNT + 1)
```
Comment (abalakh) [inline tests/foreman/cli/test_repository.py]: and here

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_repository.py -m file_repo
S...
----------------------------------------------------------------------
Ran 4 tests in 92.507s

OK (SKIP=1)
```

**Pull request merged**